zap/src/main/java/org/parosproxy/paros/network/HttpSender.java

```diff
@@ -237,6 +240,16 @@ public HttpSender(ConnectionParam connectionParam, boolean useGlobalState, int i
                         defaultUserAgent);
 
         setUseGlobalState(useGlobalState);
+        setUseCookies(true);
+    }
+
+    public HttpSender(
```

Comment:
It's not necessary to add another constructor (we should move to builder pattern), instead the calling code can make use of the cookie setter after creating the instance.